### ISSUE TITLE
OpenWPM hangs

### ISSUE THREAD

**Issue by MaazBinMusa** [openwpm/OpenWPM#957]
In the deploy firefox file I install an add_on adblocker towards the end, after openwpm xpi is installed.

After this I run a simple crawl of 50 websites. This results in the crawl getting executed but gives these warning (about 50 of these) and the demo.py does not shutdown. It Just hangs. 

```
storage_controller   - INFO     - Awaiting all tasks for visit_id 2394433868866261
sql_provider         - WARNING  - Visit with visit_id 2394433868866261 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 8857879637229978
sql_provider         - WARNING  - Visit with visit_id 8857879637229978 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 6605805886605677
sql_provider         - WARNING  - Visit with visit_id 6605805886605677 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 4976310008382932
sql_provider         - WARNING  - Visit with visit_id 4976310008382932 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 1221128306404878
sql_provider         - WARNING  - Visit with visit_id 1221128306404878 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 1903248132902091
sql_provider         - WARNING  - Visit with visit_id 1903248132902091 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 1374176394863563
sql_provider         - WARNING  - Visit with visit_id 1374176394863563 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 2714861535115475
sql_provider         - WARNING  - Visit with visit_id 2714861535115475 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 5099912865844243
sql_provider         - WARNING  - Visit with visit_id 5099912865844243 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 4791266809356625
sql_provider         - WARNING  - Visit with visit_id 4791266809356625 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 7916683500461527
sql_provider         - WARNING  - Visit with visit_id 7916683500461527 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 7374938213921325
sql_provider         - WARNING  - Visit with visit_id 7374938213921325 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 5036639023635015
sql_provider         - WARNING  - Visit with visit_id 5036639023635015 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 3010862000196919
sql_provider         - WARNING  - Visit with visit_id 3010862000196919 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 2244968277718576
sql_provider         - WARNING  - Visit with visit_id 2244968277718576 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 319991707537304
sql_provider         - WARNING  - Visit with visit_id 319991707537304 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 3455507760516858
sql_provider         - WARNING  - Visit with visit_id 3455507760516858 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 4595049379283658
sql_provider         - WARNING  - Visit with visit_id 4595049379283658 got interrupted
storage_controller   - INFO     - Awaiting all tasks for visit_id 7847554982819993
sql_provider         - WARNING  - Visit with visit_id 7847554982819993 got interrupted
Executing <Task pending name='Task-1' coro=<StorageController._run() running at /home/data/maaz/H2/verificationRun/OpenWPM/openwpm/storage/storage_controller.py:352> wait_for=<Task pending name='CompletionQueueFeeder' coro=<StorageController.update_completion_queue() running at /home/data/maaz/H2/verificationRun/OpenWPM/openwpm/storage/storage_controller.py:322> wait_for=<Future finished result=None created at /home/maaz/miniconda3/envs/openwpm/lib/python3.9/asyncio/base_events.py:424> cb=[<TaskWakeupMethWrapper object at 0x7f54c0e7a460>()] created at /home/maaz/miniconda3/envs/openwpm/lib/python3.9/asyncio/tasks.py:361> cb=[_run_until_complete_cb() at /home/maaz/miniconda3/envs/openwpm/lib/python3.9/asyncio/base_events.py:184] created at /home/maaz/miniconda3/envs/openwpm/lib/python3.9/asyncio/base_events.py:621> took 0.901 seconds
```

Running without the xpi results in a proper shutdown.

**Comment by elliottwaissbluth** [comment]
I've had the same issue plenty of times, it prevents me from doing long scrapes without manual intervention. A fix would be greatly appreciated.

**Comment by vringar** [comment]
This seems like a duplicate of #922. 
I don't think I will have time to debug this any time soon, because I don't even know where to start on this one.
How many browsers are you running in parallel? Still 30?
Without the xpi there is no data collected, so there is no data the storage controller can get stuck on saving.

**Comment by vringar** [comment]
Just dumping my thoughts in here:

When we collect a lot of data from a lot of browsers and then tell the Storage Controller to shut down it doesn't shut down because it gets hung up on trying to mark all the unfinished visit_ids as unsuccessful
https://github.com/openwpm/OpenWPM/blob/3354bc18756dfcb3119d171177967aafb87b8d84/openwpm/storage/storage_controller.py#L250-L267

Are you using the SQLite storage provider? Or are you saving to the cloud? Or did you write your own?
